compiler/nir/cir.nim

```diff
@@ -554,6 +562,54 @@ template moveToDataSection(body: untyped) =
     c.data.add c.code[i]
   setLen c.code, oldLen
 
+proc genGccAsm(c: var GeneratedCode; t: Tree; n: NodePos) =
+  c.add "__asm__ "
+  c.add ParLe
+  c.add NewLine
+
+  var asmTemplate = true
+  var left = 0
+  var s = ""
+
+  template maybeAddQuote =
+    if asmTemplate: c.add Quote
+
+  template findSecUpdate(s: string): bool =
+    var result = false
+    for i in s:
+      if i == ':': result = true
+      if i notin {' ', '\t', '\n'}: break #found char
+    result
+
+  maybeAddQuote
+  for ch in sons(t, n):
+    if t[ch].kind == Verbatim:
+      s = c.m.lit.strings[t[ch].litId]
+      left = 0
+      for i in 0..s.high:
+        if s[i] == '\n':
+          c.add s[left..i - 1]
```

Comment:
What? Why? Didn't you write a parser for the asm? Shouldn't it split it at newlines if they are important?